drivers/i2c/i2c_ll_stm32_v2.c

```diff
@@ -118,21 +118,34 @@ static void stm32_i2c_slave_event(const struct device *dev)
 	const struct i2c_stm32_config *cfg = dev->config;
 	struct i2c_stm32_data *data = dev->data;
 	I2C_TypeDef *i2c = cfg->i2c;
-	const struct i2c_slave_callbacks *slave_cb =
-		data->slave_cfg->callbacks;
+	const struct i2c_slave_callbacks *slave_cb;
+	struct i2c_slave_config *slave_cfg;
+	uint16_t slave_address;
+
+	/* Choose the right slave from the address match code */
+	slave_address = LL_I2C_GetAddressMatchCode(i2c) >> 1;
```

Comment:
note: according to docs, `LL_I2C_GetAddressMatchCode` range of return values is `@retval Value between Min_Data=0x00 and Max_Data=0x3F`, so maybe use `uint8_t` here?

Reply:
The documentation is definitely wrong, it should be `Max_data=0xff`. That said your comment is still valid, I'll update the code.